### ISSUE TITLE
Color blocks could show the resulting color

### ISSUE THREAD

**Issue by cdsmith** [google/codeworld#326]
It was suggested that color blocks could show the resulting colors on the block itself, instead of waiting until the program is run.

**Comment by cdsmith** [comment]
This was an idea that came up during a testing session last night.  I'm not actually sure that I like it, though.  It might be a bit noisy UI-wise, and #194 might be the better approach for getting immediate feedback.

(On the other hand, you could argue that giving immediate feedback about *intermediate* values like this is an even greater opportunity than merely doing a continuous update of the final program.  You could also argue, then, for preview options for other types as well.  This would be an amazing direction to go, eventually.)

**Comment by stefan-j** [comment]
Are you referring to color blocks such as blue, green and so forth? If so maybe a picture of a paint bucket in the correct color might be a compromise?

**Comment by cdsmith** [comment]
Yes, I definitely like the idea of color *literals* showing the color.  This also has the advantage that it doesn't sound as hard.

The more interesting idea was to have color *expression* blocks - such as `mixed`, `light`, etc., also show their resulting color, computed on the fly based on their arguments.  This extrapolates to other types as well: perhaps picture-type blocks could have a mini-picture on them.  Or number blocks could show the values of intermediate expressions.

Of course: (a) this would only work for expressions without local variables, and (b) it would get very noisy, so if it were implemented, maybe it should be moved to a "Preview" option in the context menu, or some such thing.  This is certainly not a priority, but I could see it being a valuable debugging tool.

**Comment by cdsmith** [comment]
I'll split the preview thing into a separate issue, and leave this one for the literal color icons.
